lib/uplink/ext/mapping_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestMapping_Add(t *testing.T) {
+	{
+		t.Log("string")
+		testMap := newMapping()
+
+		str := "testing 123"
+		strToken := testMap.Add(str)
+
+		gotStr, ok := testMap.values[strToken]
+		require.True(t, ok)
+		assert.Equal(t, str, gotStr)
```

Comment:
I think we should use `assert` instead of `require` for assertions that don't prevent the test from finishing. This way, if there are multiple failing assertions, we will see them all at once instead of having to fix each thing and re-run the tests repeatedly.

Reply:
yeah but output of such multiple failures in one test case can be sometimes unreadable. also unit tests are fast to execute and re-running shouldn't be a problem. If I didn't convince you treat this as a nit :)

Reply:
If we're taking the fact that these are unit tests into consideration then i think it's also fair to consider that these asserts aren't in any large loops or anything and the most number of errors you'd see simultaneously is 2.